source/patterns/@aws-solutions-constructs/resources/kms-key-policy-updater/lib/index.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { KMSClient, GetKeyPolicyCommand, DescribeKeyCommand, PutKeyPolicyCommand, KeyManagerType } from "@aws-sdk/client-kms";
+
+const kmsClient = new KMSClient();
+
+export const handler = async (e: any, context: any) => {
+
+  let status = 'SUCCESS';
+  let responseData = {};
+
+  if (e.RequestType === 'Create' || e.RequestType === 'Update') {
+
+    try {
+      const kmsKeyId = e.ResourceProperties.KmsKeyId;
+      const cloudFrontDistributionId = e.ResourceProperties.CloudFrontDistributionId;
+      const accountId = e.ResourceProperties.AccountId;
+      const region = process.env.AWS_REGION;
+
+      const describeKeyCommandResponse = await kmsClient.send(new DescribeKeyCommand({
+        KeyId: kmsKeyId
+      }));
+
+      if (describeKeyCommandResponse.KeyMetadata?.KeyManager === KeyManagerType.AWS) {
```

Comment:
So what we discussed is something like this checking for the policy existing somewhere around line 51

Reply:
Added this in d50315298ecb4a072a128656ad57a38761b9aab6